File: Array/1.TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for idx in range(len(nums)):
            if nums[idx] in dic:
                return [dic[nums[idx]], idx]
            dic[target-nums[idx]] = idx
        return [0, 1]

        # if target - nums is in the dic this mean some values can adding to sum and return the result

File: Array/1.TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        for (int i = 0;i<nums.size();i++){
            if (mp[nums[i]]) {
                return {mp[nums[i]]-1, i};
            }
            // i add 1 here for special case when the pair in the first one like: {0, 5}, {0, 3} but if check 0 = false => ignore pass case so to avoid that
            mp[target - nums[i]] = i+1;
        }

        return {};
    }
};